pkg/api/cmd/server.go

```diff
@@ -529,17 +544,18 @@ func (s *server) Run(ctx context.Context, metrics metrics.Metrics, logger *zap.L
 		return fmt.Errorf("failed to create API gateway: %v", err)
 	}
 
-	if err := apiGateway.Start(ctx, gatewayHandler); err != nil {
+	serverCtx, serverCtxCancel := context.WithCancel(context.Background())
+	defer serverCtxCancel()
+	if err := apiGateway.Start(serverCtx, gatewayHandler); err != nil {
```

Comment:
Bascailly, this and the envoy preStop script was the issue.
The envoy preStop was not waiting for the api gateway to shut down gracefully because the `netstat` command was removed from envoy image.

Also, the was an issue in the api container using the app context, so at the moment the app gets the sigterm, it starts sending GOAWAY frames to client, which cases 499 errors. we must use a different ctx so we can manage the shut down gracefully.